src/components/Functions/axisFormat.ts

```diff
@@ -11,10 +11,21 @@ export const formatDateAsDayMonth = (i: string): string => {
     return `${parts[1]}/${parts[2]}`;
 }
 
-export const formatNumberAsK = (n: string | number = 0): string =>
-    (Math.abs(+n) > 1000)
-        ? `${(+n / 1000).toFixed(1)}k`
-        : `${(+n).toFixed(0)}`;
+export const formatNumberAsK = (n: string | number = 0): string => {
+    let divisor = 1;
+    let suffix = '';
+    if (Math.abs(+n) > 1000000) {
+        divisor = 1000000;
+        suffix = 'M';
+    } else if (Math.abs(+n) > 1000) {
+        divisor = 1000;
+        suffix = 'k';
+    }
+
+    const output = +n / divisor;
+    const remainLen = 3 - `${output.toFixed(0)}`.length + (output > 0 ? 0 : 1);
```

Comment:
I now understand the number `3` at the start of the row, can we move it to a constant and name it appropriately so it is easier to read?